consumer/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/consumer/service/oppgavebehandling/RestOppgaveBehandlingServiceImpl.kt

```diff
@@ -0,0 +1,447 @@
+                endretTidspunkt = endretTidspunkt
+        )
+    }
+
+    val leggTilbakeOppgaveDelegate: LeggTilbakeOppgaveDelegate = LeggTilbakeOppgaveDelegate(this, arbeidsfordelingService)
+    private val log = LoggerFactory.getLogger(RestOppgaveBehandlingServiceImpl::class.java)
+
+
+    override fun opprettOppgave(request: OpprettOppgaveRequest): OpprettOppgaveResponse {
+        val behandling: Optional<Behandling> = kodeverksmapperService.mapUnderkategori(request.underkategoriKode)
+        val oppgaveTypeMapped: String = kodeverksmapperService.mapOppgavetype(request.oppgavetype)
+        val aktorId = getAktorId(request.fnr)
+        if (aktorId == null || aktorId.isEmpty()) {
+            throw Exception("AktørId-mangler på person")
+        }
+        val response = apiClient.opprettOppgave(
+                authorization = stsService.systemUserToken,
+                xminusCorrelationMinusID = correlationId(),
+                postOppgaveRequestJsonDTO = PostOppgaveRequestJsonDTO(
+                        opprettetAvEnhetsnr = request.opprettetavenhetsnummer,
+                        aktoerId = aktorId,
+                        behandlesAvApplikasjon = request.behandlesAvApplikasjon,
+                        tilordnetRessurs = request.ansvarligIdent,
+                        tildeltEnhetsnr = request.ansvarligEnhetId,
+                        beskrivelse = request.beskrivelse,
+                        temagruppe = request.temagruppe,
+                        tema = request.tema,
+                        behandlingstema = behandling.map(Behandling::getBehandlingstema).orElse(""),
+                        oppgavetype = oppgaveTypeMapped,
+                        behandlingstype = behandling.map(Behandling::getBehandlingstype).orElse(""),
+                        aktivDato = LocalDate.now(),
+                        fristFerdigstillelse = request.oppgaveFrist,
+                        prioritet = PostOppgaveRequestJsonDTO.Prioritet.valueOf(stripTemakode(request.prioritet))
+                )
+        )
+        return OpprettOppgaveResponse(response.id?.toString() ?: throw RuntimeException("No oppgaveId found"))
+    }
+
+    override fun opprettSkjermetOppgave(request: OpprettOppgaveRequest): OpprettOppgaveResponse {
+        val behandling: Optional<Behandling> = kodeverksmapperService.mapUnderkategori(request.underkategoriKode)
+        val oppgaveTypeMapped: String = kodeverksmapperService.mapOppgavetype(request.oppgavetype)
+        val aktorId = getAktorId(request.fnr)
+        if (aktorId == null || aktorId.isEmpty()) {
+            throw Exception("AktørId-mangler på person")
+        }
+        val response = apiClient.opprettOppgave(
+                authorization = stsService.systemUserToken,
+                xminusCorrelationMinusID = correlationId(),
+                postOppgaveRequestJsonDTO = PostOppgaveRequestJsonDTO(
+                        aktoerId = aktorId,
```

Comment:
nesten samme code i opprettSkjermetOppgave og  method opprettOppgave

Reply:
OpprettOppgave og skjermetOppgave  har ein del forskjell i oppførsel, som då kan lage problemer/ unødig kompleiksistet om dei blir slått saman
Skjermet oppgave forholder seg ikkje til tildelt enhet og arbeidsfordeling da dette er oppgave sitt område, mens opprettOppgave håndterer både kvar oppgåva skal og kven som skal ha den. 
Det som kan gjerast er å lag ein gjorsporring-funksjon som begge metodene bruker for det aktuelle kallet mot oppgave, det trenger ikkje å forholde seg til oppgaveobjektet, kun at det er eit oppgave-objekt og at det skal vere visse metadata i headeren som går over.